cap-primitives/src/rustix/fs/dir_utils.rs

```diff
@@ -100,15 +100,13 @@ pub(crate) fn canonicalize_options() -> OpenOptions {
 /// This function is not sandboxed and may trivially access any path that the
 /// host process has access to.
 pub(crate) fn open_ambient_dir_impl(path: &Path, _: AmbientAuthority) -> io::Result<fs::File> {
-    // This is for `std::fs`, so we don't have `dir_required`, so set
-    // `O_DIRECTORY` manually.
-    let flags = OFlags::DIRECTORY | target_o_path();
-
     let mut options = fs::OpenOptions::new();
     options.read(true);
 
     #[cfg(not(target_os = "wasi"))]
-    options.custom_flags(flags.bits() as i32);
+    options.custom_flags((OFlags::DIRECTORY | target_o_path()).bits() as i32);
+    #[cfg(target_os = "wasi")]
+    options.directory(true);
```

Comment:
Looks good. Can you preserve the comment about how the reason we need to do this is that we're calling std::fs which doesn't have a `dir_required` function?

Reply:
done